Worksheet_D/Worksheett_D.py

```diff
@@ -0,0 +1,115 @@
+from __builtin__ import xrange
+
+
+ class OxoBoard:
+     def __init__(self):
+         """ The initialiser. Initialise any fields you need here. """
+         self.oxoboard = {(0, 0): 0, (1, 0): 0, (2, 0): 0, (0, 1): 0, (1, 1): 0, (2, 1): 0, (0, 2): 0, (1, 2): 0,
+                          (2, 2): 0}
+
+         def get_square(self, x, y):
+             """ Return 0, 1 or 2 depending on the contents of the specified square. """
+         self.x = x
+         self.y = y
+         return self.oxoboard[x,y]
+
+
+     def set_square(self, x, y, mark):
+         """ If the specified square is currently empty (0), fill it with mark and return True.
+             If the square is not empty, leave it as-is and return False. """
+         if self.oxoboard[x,y] == 0:
+             self.oxoboard[x,y] = mark
+             return True
+         else:
+             return False
+
+
+         raise NotImplementedError("TODO: implement set_square")
+
+     def is_board_full(self):
+         """ If there are still empty squares on the board, return False.
+             If there are no empty squares, return True. """
+         Board = 9
+         for i in xrange(0, 3):
+             for j in yrange(0, 3):
+                 if self.board[i, j] > 0:
+                     Board -= 1
+                     if Board == 0:
+                         return True
+                     return False

```

Comment:
This will cause the function to return `False` on the first non-empty square it encounters. Or did you intend for this to be outside the `for`-`for`-`if` block (in which case this is an indentation error)?